packages/frontend/src/components/composer/menuAttachment.tsx

```diff
@@ -36,28 +39,56 @@ export default function MenuAttachment({
   const openConfirmationDialog = useConfirmationDialog()
   const { sendVideoChatInvitation } = useVideoChat()
   const { openDialog, closeDialog } = useDialog()
+  const { sendMessage } = useMessage()
   const [settings] = useStore(SettingsStoreInstance)
   const accountId = selectedAccountId()
 
+  const sendMultipleFiles = (filePaths: string[], msgViewType: Viewtype) => {
+    if (!selectedChat) {
+      throw new Error('no chat selected')
+    }
+    openDialog(ConfirmSendingFiles, {
+      sanitizedFileList: filePaths.map(path => ({ name: basename(path) })),
+      chatName: selectedChat.name,
+      onClick: async (isConfirmed: boolean) => {
+        if (!isConfirmed) {
+          return
+        }
+
+        for (const filePath of filePaths) {
+          sendMessage(accountId, selectedChat.id, {
+            file: filePath,
+            viewtype: msgViewType,
+          }).then(() => {
+            // start sending other files, don't wait until last file is sent
+            runtime.removeTempFile(filePath)
```

Comment:
So this is copy-pasted from

https://github.com/deltachat/deltachat-desktop/blob/f0e4cc5ef0a8177876b322cb0753ee04b7cdece1/packages/frontend/src/components/message/MessageListAndComposer.tsx#L168-L190

but in that code we also do `writeTempFileFromFile`. Why do we have to do that there, but not here.

And, most importantly, if we decide to not create temp files, we should not `runtime.removeTempFile()` here. Currently in Electron this simply gives an error of "Path is outside of the temp folder". I did not check what the browser version does, but it [does not look good](https://github.com/deltachat/deltachat-desktop/blob/5f640db6f6b809f9ffa05822045bf13519cac3a2/packages/target-browser/src/backendApi.ts#L118-L138).

I think it'd be nice to add a docstring to `removeTempFile`, or somehow make it safer in general.

Reply:
> but in that code we also do writeTempFileFromFile. Why do we have to do that there, but not here.

because here we get valid file paths.
everything would be cleaner if we could pass files directly to core without these temp files, but that api doesn't exist yet.

For drag and drop I needed to change it, because it was an electron specific api and also did not always work #3595.

> but it [does not look good](https://github.com/deltachat/deltachat-desktop/blob/5f640db6f6b809f9ffa05822045bf13519cac3a2/packages/target-browser/src/backendApi.ts#L118-L138).

I know, needs to be improved, I plan to move the used temp folder to a different place, I already added a todo item to #4232 - in that process we can make the path check even safer.

Reply:
> I know, needs to be improved

I'm interested in a sense that it wouldn't work great if you pass normal, non-temporary files there 😅